gt/main.c

```diff
@@ -900,6 +1133,24 @@ gt_proc(void *arg)
 			ETHER_TYPE_IPv4);
 		process_pkts_acl(&gt_conf->net->front, lcore, &acl6,
 			ETHER_TYPE_IPv6);
+
+		if (cur_tsc - last_tsc >= frag_scan_timeout_cycles) {
+			death_row.cnt = 0;
+			ret = rte_ip_frag_table_iterate(
+				instance->frag_tbl, &death_row, &next);
+			if (ret != 0) {
+				RTE_LOG(WARNING, GATEKEEPER,
+					"gt: failed to call rte_ip_frag_table_iterate() to iterate over the fragmentation table at lcore %u\n",
+					lcore);
+			}
+
+			/* Process the death packets. */
+			process_death_row(socket, port,
+				tx_queue, 1, &death_row,
+				instance, gt_conf);
+
+			last_tsc = cur_tsc;
```

Comment:
Since `cur_tsc` was obtained, we processed a batch of packets and scanned a bucket of the fragment table. In other words, `cur_tsc` is quite old at this point in the code. Call `rte_rdtsc()` to update `last_tsc`.